src/material/paginator/testing/paginator-harness.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessPredicate} from '@angular/cdk/testing';
+import {MatSelectHarness} from '@angular/material/select/testing';
+import {PaginatorHarnessFilters} from './paginator-harness-filters';
+
+
+/** Harness for interacting with a standard mat-paginator in tests. */
+export class MatPaginatorHarness extends ComponentHarness {
```

Comment:
Note that I've only added setter methods here. I avoided setting up any getters, because we don't have reliable ways to get all of the values out of the DOM. E.g. the `MatPaginator.length` is available in the DOM, but it's localized so we can't rely on it.

Reply:
What if we make the APIs about the page size _option_ instead of treating it as a number? E.g. something like 
`getSelectedPageSizeOption` and `selectPageSizeOption`?

Reply:
We can actually retrieve the page size somewhat reliably, because we put it directly in the DOM. What I meant above is for the other inputs like `length` and `pageIndex` which are only available in a string that looks like `10 of 50`.

Reply:
Oh yeah, I totally spaces out on your comment. I think we could do one or both of two options:
1. Just have an API that returns that whole localized string
2. Have an API like `getPageIndex` / `getLength` that throw if they're not in exactly the form `\d+ of \d+`

Reply:
I went with option 1 since it allows consumers to handle localized strings if they want to.